packages/react/src/Button/ButtonBase.tsx

```diff
@@ -64,45 +84,65 @@ const ButtonBase = forwardRef(
     }
 
     return (
-      <StyledButton
-        as={Component}
-        sx={sxStyles}
-        {...rest}
-        ref={innerRef}
-        data-block={block ? 'block' : null}
-        data-size={size === 'small' || size === 'large' ? size : undefined}
-        data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
-        data-inactive={inactive ? true : undefined}
-      >
-        {Icon ? (
-          <Icon />
-        ) : (
-          <>
-            <Box as="span" data-component="buttonContent" sx={getAlignContentSize(alignContent)}>
-              {LeadingVisual && (
-                <Box as="span" data-component="leadingVisual" sx={{...iconWrapStyles}}>
-                  <LeadingVisual />
+      <ConditionalWrapper if={Boolean(loading)} sx={{display: block ? 'block' : 'inline-block'}} data-loading-wrapper>
+        <StyledButton
+          as={Component}
+          sx={sxStyles}
+          aria-disabled={loading ? true : undefined}
+          {...rest}
+          ref={innerRef}
+          data-block={block ? 'block' : null}
+          data-inactive={inactive ? true : undefined}
+          data-loading={Boolean(loading)}
+          data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
+          data-size={size === 'small' || size === 'large' ? size : undefined}
+          aria-describedby={[loadingAnnouncementID, ariaDescribedBy]
+            .filter(descriptionID => Boolean(descriptionID))
+            .join(' ')}
+          // aria-labelledby is needed because the accessible name becomes unset when the button is in a loading state.
+          // We only set it when the button is in a loading state because it will supercede the aria-label when the screen
+          // reader announces the button name.
+          aria-labelledby={loading ? `${uuid}-label` : ariaLabelledBy}
```

Comment:
I'm wondering if we could add `ariaLabelledBy` to the first condition too? 

(i.e. `${uuid}-label ${ariaLabelledBy}` : `ariaLabelledBy`)

This allows instances that rely on `aria-labelledby` to remain covered, even if the button is in a loading state.

Reply:
Oh I didn't realize we could have 2 values in aria-labelledby. That could work! I'll give it a try...